### PR TITLE
Verious performance improvments

### PR DESCRIPTION
**Pull request by crspeller**
- Part 1. More to come.

### CODE REVIEW

File: webapp/stores/error_store.jsx [hunk 1]
```diff
@@ -70,7 +70,9 @@ class ErrorStoreClass extends EventEmitter {
 
         BrowserStore.removeGlobalItem('last_error');
         BrowserStore.removeGlobalItem('last_error_conn');
-        this.emitChange();
+        if (lastError) {
```
Comment (coreyhulen) [inline webapp/stores/error_store.jsx]: I think this is how the error bar disappears on websocket reconnect.

File: webapp/utils/async_client.jsx [hunk 2]
```diff
@@ -131,7 +131,6 @@ export function updateLastViewedAt(id) {
         () => {
             callTracker.updateLastViewed = 0;
             ErrorStore.clearLastError();
-            ErrorStore.emitChange();
```
Comment (coreyhulen) [inline webapp/utils/async_client.jsx]: I think this is how the error bar disappears on server reconnect.

Comment (coreyhulen) [inline webapp/utils/async_client.jsx]: updateLastView is called every 30s so if the server is down the error will appear.  When the server is backup the error should disappear on next successful reconnect.

Comment (crspeller) [inline webapp/utils/async_client.jsx]: Yes. The emitChange function is called within the clearLastError function. We where double calling the function.

File: webapp/stores/error_store.jsx (hunk 1 above)
Comment (crspeller) [inline webapp/stores/error_store.jsx]: Yes. This is just a guard so that we only bother to emit a clear message if there was actually an error to clear.

**Comment by coreyhulen** [comment]
+1

**Comment by enahum** [comment]
+1

**Comment by jwilander** [comment]
+1

**Pull request merged**